python/ray/_private/log_monitor.py

```diff
@@ -533,31 +536,82 @@ def is_proc_alive(pid):
+        help="Specify the max bytes for rotating log file.",
     )
     parser.add_argument(
         "--logging-rotate-backup-count",
-        required=False,
+        required=True,
         type=int,
-        default=ray_constants.LOGGING_ROTATE_BACKUP_COUNT,
-        help="Specify the backup count of rotated log file, default is "
-        f"{ray_constants.LOGGING_ROTATE_BACKUP_COUNT}.",
+        help="Specify the backup count of rotated log file.",
     )
+    parser.add_argument(
+        "--stdout-filepath",
+        required=False,
+        default="",
+        type=str,
+        help="The filepath to dump log monitor stdout.",
+    )
+    parser.add_argument(
+        "--stderr-filepath",
+        required=False,
+        default="",
+        type=str,
+        help="The filepath to dump log monitor stderr.",
+    )
+
     args = parser.parse_args()
-    setup_component_logger(
+
+    # Disable log rotation for windows platform.
+    logging_rotation_bytes = args.logging_rotate_bytes if sys.platform != "win32" else 0
+    logging_rotation_backup_count = (
+        args.logging_rotate_backup_count if sys.platform != "win32" else 1
+    )
+    logging_params = dict(
         logging_level=args.logging_level,
         logging_format=args.logging_format,
         log_dir=args.logs_dir,
         filename=args.logging_filename,
-        max_bytes=args.logging_rotate_bytes,
-        backup_count=args.logging_rotate_backup_count,
+        max_bytes=logging_rotation_bytes,
+        backup_count=logging_rotation_backup_count,
     )
 
+    with open("/tmp/debug-monitor-log.debug", "a+") as file:
+        file.write(f"logging params = {logging_params}\n")
+        file.flush()
+
+    logger = setup_component_logger(**logging_params)
+
+    # Setup stdout/stderr redirect files if redirection enabled
+    if args.stdout_filepath:
+        StreamRedirector.redirect_stdout(
+            args.stdout_filepath,
+            logging_rotation_bytes,
+            logging_rotation_backup_count,
+            False,
+            False,
+        )
+    if args.stderr_filepath:
+        StreamRedirector.redirect_stderr(
+            args.stderr_filepath,
+            logging_rotation_bytes,
+            logging_rotation_backup_count,
+            False,
+            False,
+        )
+
+    # Setup stdout/stderr redirect files
+    stdout_fileno = sys.stdout.fileno()
+    stderr_fileno = sys.stderr.fileno()
+    # We also manually set sys.stdout and sys.stderr because that seems to
+    # have an effect on the output buffering. Without doing this, stdout
+    # and stderr are heavily buffered resulting in seemingly lost logging
+    # statements. We never want to close the stdout file descriptor, dup2 will
+    # close it when necessary and we don't want python's GC to close it.
+    sys.stdout = open_log(stdout_fileno, unbuffered=True, closefd=False)
+    sys.stderr = open_log(stderr_fileno, unbuffered=True, closefd=False)
```

Comment:
This happens multiple times in the code base, lets make a shared function

Reply:
I understand, can I do it when all components rotated and merged?

Reply:
sure